File: app/src/main/java/uk/co/ourfriendirony/medianotifier/clients/rawg/game/get/GameGet.java
package uk.co.ourfriendirony.medianotifier.clients.rawg.game.get;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import java.util.Date;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
        "id",
        "slug",
        "name",
        "name_original",
        "description",
        "metacritic",
        "metacritic_platforms",
        "released",
        "tba",
        "updated",
        "background_image",
        "background_image_additional",
        "website",
        "rating",
        "rating_top",
        "ratings",
        "reactions",
        "added",
        "added_by_status",
        "playtime",
        "screenshots_count",
        "movies_count",
        "creators_count",
        "achievements_count",
        "parent_achievements_count",
        "reddit_url",
        "reddit_name",
        "reddit_description",
        "reddit_logo",
        "reddit_count",
        "twitch_count",
        "youtube_count",
        "reviews_text_count",
        "ratings_count",
        "suggestions_count",
        "alternative_names",
        "metacritic_url",
        "parents_count",
        "additions_count",
        "game_series_count",
        "user_game",
        "reviews_count",
        "saturated_color",
        "dominant_color",
        "parent_platforms",
        "platforms",
        "stores",
        "developers",
        "genres",
        "tags",
        "publishers",
        "esrb_rating",
        "clip",
        "description_raw"
})
public class GameGet {

    @JsonProperty("id")
    private Integer id;
    @JsonProperty("slug")
    private String slug;
    @JsonProperty("name")
    private String name;
    @JsonProperty("name_original")
    private String nameOriginal;
    @JsonProperty("description")
    private String description;
    @JsonProperty("metacritic")
    private Integer metacritic;
    @JsonProperty("metacritic_platforms")
    private List<Object> metacriticPlatforms = null;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    @JsonProperty("released")
    private Date released;
    @JsonProperty("tba")
    private Boolean tba;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    @JsonProperty("updated")
    private Date updated;
    @JsonProperty("background_image")
    private String backgroundImage;
    @JsonProperty("background_image_additional")
    private String backgroundImageAdditional;
    @JsonProperty("website")
    private String website;
    @JsonProperty("rating")
    private Double rating;
    @JsonProperty("rating_top")
    private Integer ratingTop;
    @JsonProperty("ratings")
    private List<GameGetRating> ratings = null;
    @JsonProperty("reactions")
    private GameGetReactions reactions;
    @JsonProperty("added")
    private Integer added;
    @JsonProperty("added_by_status")
    private GameGetAddedByStatus addedByStatus;
    @JsonProperty("playtime")
    private Integer playtime;
    @JsonProperty("screenshots_count")
    private Integer screenshotsCount;
    @JsonProperty("movies_count")
    private Integer moviesCount;
    @JsonProperty("creators_count")
    private Integer creatorsCount;
    @JsonProperty("achievements_count")
    private Integer achievementsCount;
    @JsonProperty("parent_achievements_count")
    private Integer parentAchievementsCount;
    @JsonProperty("reddit_url")
    private String redditUrl;
    @JsonProperty("reddit_name")
    private String redditName;
    @JsonProperty("reddit_description")
    private String redditDescription;
    @JsonProperty("reddit_logo")
    private String redditLogo;
    @JsonProperty("reddit_count")
    private Integer redditCount;
    @JsonProperty("twitch_count")
    private Integer twitchCount;
    @JsonProperty("youtube_count")
    private Integer youtubeCount;
    @JsonProperty("reviews_text_count")
    private Integer reviewsTextCount;
    @JsonProperty("ratings_count")
    private Integer ratingsCount;
    @JsonProperty("suggestions_count")
    private Integer suggestionsCount;
    @JsonProperty("alternative_names")
    private List<String> alternativeNames = null;
    @JsonProperty("metacritic_url")
    private String metacriticUrl;
    @JsonProperty("parents_count")
    private Integer parentsCount;
    @JsonProperty("additions_count")
    private Integer additionsCount;
    @JsonProperty("game_series_count")
    private Integer gameSeriesCount;
    @JsonProperty("user_game")
    private Object userGame;
    @JsonProperty("reviews_count")
    private Integer reviewsCount;
    @JsonProperty("saturated_color")
    private String saturatedColor;
    @JsonProperty("dominant_color")
    private String dominantColor;
    @JsonProperty("parent_platforms")
    private List<GameGetParentPlatformGroup> parentPlatforms = null;
    @JsonProperty("platforms")
    private List<GameGetPlatformGroup> platforms = null;
    @JsonProperty("stores")
    private List<GameGetStoreGroup> stores = null;
    @JsonProperty("developers")
    private List<GameGetDeveloper> developers = null;
    @JsonProperty("genres")
    private List<GameGetGenre> genres = null;
    @JsonProperty("tags")
    private List<GameGetTag> tags = null;
    @JsonProperty("publishers")
    private List<GameGetPublisher> publishers = null;
    @JsonProperty("esrb_rating")
    private GameGetEsrbRating esrbRating;
    @JsonProperty("clip")
    private Object clip;
    @JsonProperty("description_raw")
    private String descriptionRaw;
    @JsonIgnore
    private final Map<String, Object> additionalProperties = new HashMap<String, Object>();

    @JsonProperty("id")
    public Integer getId() {
        return id;
    }

    @JsonProperty("id")
    public void setId(Integer id) {
        this.id = id;
    }

    @JsonProperty("slug")
    public String getSlug() {
        return slug;
    }

    @JsonProperty("slug")
    public void setSlug(String slug) {
        this.slug = slug;
    }

    @JsonProperty("name")
    public String getName() {
        return name;
    }

    @JsonProperty("name")
    public void setName(String name) {
        this.name = name;
    }

    @JsonProperty("name_original")
    public String getNameOriginal() {
        return nameOriginal;
    }

    @JsonProperty("name_original")
    public void setNameOriginal(String nameOriginal) {
        this.nameOriginal = nameOriginal;
    }

    @JsonProperty("description")
    public String getDescription() {
        return description;
    }

    @JsonProperty("description")
    public void setDescription(String description) {
        this.description = description;
    }

    @JsonProperty("metacritic")
    public Integer getMetacritic() {
        return metacritic;
    }

    @JsonProperty("metacritic")
    public void setMetacritic(Integer metacritic) {
        this.metacritic = metacritic;
    }

    @JsonProperty("metacritic_platforms")
    public List<Object> getMetacriticPlatforms() {
        return metacriticPlatforms;
    }

    @JsonProperty("metacritic_platforms")
    public void setMetacriticPlatforms(List<Object> metacriticPlatforms) {
        this.metacriticPlatforms = metacriticPlatforms;
    }

    @JsonProperty("released")
    public Date getReleased() {
        return released;
    }

    @JsonProperty("released")
    public void setReleased(Date released) {
        this.released = released;
    }

    @JsonProperty("tba")
    public Boolean getTba() {
        return tba;
    }

    @JsonProperty("tba")
    public void setTba(Boolean tba) {
        this.tba = tba;
    }

    @JsonProperty("updated")
    public Date getUpdated() {
        return updated;
    }

    @JsonProperty("updated")
    public void setUpdated(Date updated) {
        this.updated = updated;
    }

    @JsonProperty("background_image")
    public String getBackgroundImage() {
        return backgroundImage;
    }

    @JsonProperty("background_image")
    public void setBackgroundImage(String backgroundImage) {
        this.backgroundImage = backgroundImage;
    }

    @JsonProperty("background_image_additional")
    public String getBackgroundImageAdditional() {
        return backgroundImageAdditional;
    }

    @JsonProperty("background_image_additional")
    public void setBackgroundImageAdditional(String backgroundImageAdditional) {
        this.backgroundImageAdditional = backgroundImageAdditional;
    }

    @JsonProperty("website")
    public String getWebsite() {
        return website;
    }

    @JsonProperty("website")
    public void setWebsite(String website) {
        this.website = website;
    }

    @JsonProperty("rating")
    public Double getRating() {
        return rating;
    }

    @JsonProperty("rating")
    public void setRating(Double rating) {
        this.rating = rating;
    }

    @JsonProperty("rating_top")
    public Integer getRatingTop() {
        return ratingTop;
    }

    @JsonProperty("rating_top")
    public void setRatingTop(Integer ratingTop) {
        this.ratingTop = ratingTop;
    }

    @JsonProperty("ratings")
    public List<GameGetRating> getRatings() {
        return ratings;
    }

    @JsonProperty("ratings")
    public void setRatings(List<GameGetRating> ratings) {
        this.ratings = ratings;
    }

    @JsonProperty("reactions")
    public GameGetReactions getReactions() {
        return reactions;
    }

    @JsonProperty("reactions")
    public void setReactions(GameGetReactions reactions) {
        this.reactions = reactions;
    }

    @JsonProperty("added")
    public Integer getAdded() {
        return added;
    }

    @JsonProperty("added")
    public void setAdded(Integer added) {
        this.added = added;
    }

    @JsonProperty("added_by_status")
    public GameGetAddedByStatus getAddedByStatus() {
        return addedByStatus;
    }

    @JsonProperty("added_by_status")
    public void setAddedByStatus(GameGetAddedByStatus addedByStatus) {
        this.addedByStatus = addedByStatus;
    }

    @JsonProperty("playtime")
    public Integer getPlaytime() {
        return playtime;
    }

    @JsonProperty("playtime")
    public void setPlaytime(Integer playtime) {
        this.playtime = playtime;
    }

    @JsonProperty("screenshots_count")
    public Integer getScreenshotsCount() {
        return screenshotsCount;
    }

    @JsonProperty("screenshots_count")
    public void setScreenshotsCount(Integer screenshotsCount) {
        this.screenshotsCount = screenshotsCount;
    }

    @JsonProperty("movies_count")
    public Integer getMoviesCount() {
        return moviesCount;
    }

    @JsonProperty("movies_count")
    public void setMoviesCount(Integer moviesCount) {
        this.moviesCount = moviesCount;
    }

    @JsonProperty("creators_count")
    public Integer getCreatorsCount() {
        return creatorsCount;
    }

    @JsonProperty("creators_count")
    public void setCreatorsCount(Integer creatorsCount) {
        this.creatorsCount = creatorsCount;
    }

    @JsonProperty("achievements_count")
    public Integer getAchievementsCount() {
        return achievementsCount;
    }

    @JsonProperty("achievements_count")
    public void setAchievementsCount(Integer achievementsCount) {
        this.achievementsCount = achievementsCount;
    }

    @JsonProperty("parent_achievements_count")
    public Integer getParentAchievementsCount() {
        return parentAchievementsCount;
    }

    @JsonProperty("parent_achievements_count")
    public void setParentAchievementsCount(Integer parentAchievementsCount) {
        this.parentAchievementsCount = parentAchievementsCount;
    }

    @JsonProperty("reddit_url")
    public String getRedditUrl() {
        return redditUrl;
    }

    @JsonProperty("reddit_url")
    public void setRedditUrl(String redditUrl) {
        this.redditUrl = redditUrl;
    }

    @JsonProperty("reddit_name")
    public String getRedditName() {
        return redditName;
    }

    @JsonProperty("reddit_name")
    public void setRedditName(String redditName) {
        this.redditName = redditName;
    }

    @JsonProperty("reddit_description")
    public String getRedditDescription() {
        return redditDescription;
    }

    @JsonProperty("reddit_description")
    public void setRedditDescription(String redditDescription) {
        this.redditDescription = redditDescription;
    }

    @JsonProperty("reddit_logo")
    public String getRedditLogo() {
        return redditLogo;
    }

    @JsonProperty("reddit_logo")
    public void setRedditLogo(String redditLogo) {
        this.redditLogo = redditLogo;
    }

    @JsonProperty("reddit_count")
    public Integer getRedditCount() {
        return redditCount;
    }

    @JsonProperty("reddit_count")
    public void setRedditCount(Integer redditCount) {
        this.redditCount = redditCount;
    }

    @JsonProperty("twitch_count")
    public Integer getTwitchCount() {
        return twitchCount;
    }

    @JsonProperty("twitch_count")
    public void setTwitchCount(Integer twitchCount) {
        this.twitchCount = twitchCount;
    }

    @JsonProperty("youtube_count")
    public Integer getYoutubeCount() {
        return youtubeCount;
    }

    @JsonProperty("youtube_count")
    public void setYoutubeCount(Integer youtubeCount) {
        this.youtubeCount = youtubeCount;
    }

    @JsonProperty("reviews_text_count")
    public Integer getReviewsTextCount() {
        return reviewsTextCount;
    }

    @JsonProperty("reviews_text_count")
    public void setReviewsTextCount(Integer reviewsTextCount) {
        this.reviewsTextCount = reviewsTextCount;
    }

    @JsonProperty("ratings_count")
    public Integer getRatingsCount() {
        return ratingsCount;
    }

    @JsonProperty("ratings_count")
    public void setRatingsCount(Integer ratingsCount) {
        this.ratingsCount = ratingsCount;
    }

    @JsonProperty("suggestions_count")
    public Integer getSuggestionsCount() {
        return suggestionsCount;
    }

    @JsonProperty("suggestions_count")
    public void setSuggestionsCount(Integer suggestionsCount) {
        this.suggestionsCount = suggestionsCount;
    }

    @JsonProperty("alternative_names")
    public List<String> getAlternativeNames() {
        return alternativeNames;
    }

    @JsonProperty("alternative_names")
    public void setAlternativeNames(List<String> alternativeNames) {
        this.alternativeNames = alternativeNames;
    }

    @JsonProperty("metacritic_url")
    public String getMetacriticUrl() {
        return metacriticUrl;
    }

    @JsonProperty("metacritic_url")
    public void setMetacriticUrl(String metacriticUrl) {
        this.metacriticUrl = metacriticUrl;
    }

    @JsonProperty("parents_count")
    public Integer getParentsCount() {
        return parentsCount;
    }

    @JsonProperty("parents_count")
    public void setParentsCount(Integer parentsCount) {
        this.parentsCount = parentsCount;
    }

    @JsonProperty("additions_count")
    public Integer getAdditionsCount() {
        return additionsCount;
    }

    @JsonProperty("additions_count")
    public void setAdditionsCount(Integer additionsCount) {
        this.additionsCount = additionsCount;
    }

    @JsonProperty("game_series_count")
    public Integer getGameSeriesCount() {
        return gameSeriesCount;
    }

    @JsonProperty("game_series_count")
    public void setGameSeriesCount(Integer gameSeriesCount) {
        this.gameSeriesCount = gameSeriesCount;
    }

    @JsonProperty("user_game")
    public Object getUserGame() {
        return userGame;
    }

    @JsonProperty("user_game")
    public void setUserGame(Object userGame) {
        this.userGame = userGame;
    }

    @JsonProperty("reviews_count")
    public Integer getReviewsCount() {
        return reviewsCount;
    }

    @JsonProperty("reviews_count")
    public void setReviewsCount(Integer reviewsCount) {
        this.reviewsCount = reviewsCount;
    }

    @JsonProperty("saturated_color")
    public String getSaturatedColor() {
        return saturatedColor;
    }

    @JsonProperty("saturated_color")
    public void setSaturatedColor(String saturatedColor) {
        this.saturatedColor = saturatedColor;
    }

    @JsonProperty("dominant_color")
    public String getDominantColor() {
        return dominantColor;
    }

    @JsonProperty("dominant_color")
    public void setDominantColor(String dominantColor) {
        this.dominantColor = dominantColor;
    }

    @JsonProperty("parent_platforms")
    public List<GameGetParentPlatformGroup> getParentPlatforms() {
        return parentPlatforms;
    }

    @JsonProperty("parent_platforms")
    public void setParentPlatforms(List<GameGetParentPlatformGroup> parentPlatforms) {
        this.parentPlatforms = parentPlatforms;
    }

    @JsonProperty("platforms")
    public List<GameGetPlatformGroup> getPlatforms() {
        return platforms;
    }

    @JsonProperty("platforms")
    public void setPlatforms(List<GameGetPlatformGroup> platforms) {
        this.platforms = platforms;
    }

    @JsonProperty("stores")
    public List<GameGetStoreGroup> getStores() {
        return stores;
    }

    @JsonProperty("stores")
    public void setStores(List<GameGetStoreGroup> stores) {
        this.stores = stores;
    }

    @JsonProperty("developers")
    public List<GameGetDeveloper> getDevelopers() {
        return developers;
    }

    @JsonProperty("developers")
    public void setDevelopers(List<GameGetDeveloper> developers) {
        this.developers = developers;
    }

    @JsonProperty("genres")
    public List<GameGetGenre> getGenres() {
        return genres;
    }

    @JsonProperty("genres")
    public void setGenres(List<GameGetGenre> genres) {
        this.genres = genres;
    }

    @JsonProperty("tags")
    public List<GameGetTag> getTags() {
        return tags;
    }

    @JsonProperty("tags")
    public void setTags(List<GameGetTag> tags) {
        this.tags = tags;
    }

    @JsonProperty("publishers")
    public List<GameGetPublisher> getPublishers() {
        return publishers;
    }

    @JsonProperty("publishers")
    public void setPublishers(List<GameGetPublisher> publishers) {
        this.publishers = publishers;
    }

    @JsonProperty("esrb_rating")
    public GameGetEsrbRating getEsrbRating() {
        return esrbRating;
    }

    @JsonProperty("esrb_rating")
    public void setEsrbRating(GameGetEsrbRating esrbRating) {
        this.esrbRating = esrbRating;
    }

    @JsonProperty("clip")
    public Object getClip() {
        return clip;
    }

    @JsonProperty("clip")
    public void setClip(Object clip) {
        this.clip = clip;
    }

    @JsonProperty("description_raw")
    public String getDescriptionRaw() {
        return descriptionRaw;
    }

    @JsonProperty("description_raw")
    public void setDescriptionRaw(String descriptionRaw) {
        this.descriptionRaw = descriptionRaw;
    }

    @JsonAnyGetter
    public Map<String, Object> getAdditionalProperties() {
        return this.additionalProperties;
    }

    @JsonAnySetter
    public void setAdditionalProperty(String name, Object value) {
        this.additionalProperties.put(name, value);
    }

}

File: app/src/main/java/uk/co/ourfriendirony/medianotifier/clients/rawg/game/get/GameGet.kt
package uk.co.ourfriendirony.medianotifier.clients.rawg.game.get

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonPropertyOrder
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonFormat
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.get.GameGetRating
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.get.GameGetReactions
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.get.GameGetAddedByStatus
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.get.GameGetParentPlatformGroup
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.get.GameGetPlatformGroup
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.get.GameGetStoreGroup
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.get.GameGetDeveloper
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.get.GameGetGenre
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.get.GameGetTag
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.get.GameGetPublisher
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.get.GameGetEsrbRating
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonAnyGetter
import com.fasterxml.jackson.annotation.JsonAnySetter
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.get.GameGetStore
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.get.GameGetPlatform
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.get.GameGetParentPlatform
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.search.GameSearchResult
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.search.GameSearchPlatformGroup
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.search.GameSearchStore
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.search.GameSearchRating
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.search.GameSearchAddedByStatus
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.search.GameSearchTag
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.search.GameSearchShortScreenshot
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.search.GameSearchParentPlatformGroup
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.search.GameSearchGenre
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.search.GameSearchPlatform
import uk.co.ourfriendirony.medianotifier.clients.rawg.game.search.GameSearchParentPlatform
import java.util.*

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder("id", "slug", "name", "name_original", "description", "metacritic", "metacritic_platforms", "released", "tba", "updated", "background_image", "background_image_additional", "website", "rating", "rating_top", "ratings", "reactions", "added", "added_by_status", "playtime", "screenshots_count", "movies_count", "creators_count", "achievements_count", "parent_achievements_count", "reddit_url", "reddit_name", "reddit_description", "reddit_logo", "reddit_count", "twitch_count", "youtube_count", "reviews_text_count", "ratings_count", "suggestions_count", "alternative_names", "metacritic_url", "parents_count", "additions_count", "game_series_count", "user_game", "reviews_count", "saturated_color", "dominant_color", "parent_platforms", "platforms", "stores", "developers", "genres", "tags", "publishers", "esrb_rating", "clip", "description_raw")
class GameGet {
    @get:JsonProperty("id")
    @set:JsonProperty("id")
    @JsonProperty("id")
    var id: Int? = null

    @get:JsonProperty("slug")
    @set:JsonProperty("slug")
    @JsonProperty("slug")
    var slug: String? = null

    @get:JsonProperty("name")
    @set:JsonProperty("name")
    @JsonProperty("name")
    var name: String? = null

    @get:JsonProperty("name_original")
    @set:JsonProperty("name_original")
    @JsonProperty("name_original")
    var nameOriginal: String? = null

    @get:JsonProperty("description")
    @set:JsonProperty("description")
    @JsonProperty("description")
    var description: String? = null

    @get:JsonProperty("metacritic")
    @set:JsonProperty("metacritic")
    @JsonProperty("metacritic")
    var metacritic: Int? = null

    @get:JsonProperty("metacritic_platforms")
    @set:JsonProperty("metacritic_platforms")
    @JsonProperty("metacritic_platforms")
    var metacriticPlatforms: List<Any>? = null

    @get:JsonProperty("released")
    @set:JsonProperty("released")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    @JsonProperty("released")
    var released: Date? = null

    @get:JsonProperty("tba")
    @set:JsonProperty("tba")
    @JsonProperty("tba")
    var tba: Boolean? = null

    @get:JsonProperty("updated")
    @set:JsonProperty("updated")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    @JsonProperty("updated")
    var updated: Date? = null

    @get:JsonProperty("background_image")
    @set:JsonProperty("background_image")
    @JsonProperty("background_image")
    var backgroundImage: String? = null

    @get:JsonProperty("background_image_additional")
    @set:JsonProperty("background_image_additional")
    @JsonProperty("background_image_additional")
    var backgroundImageAdditional: String? = null

    @get:JsonProperty("website")
    @set:JsonProperty("website")
    @JsonProperty("website")
    var website: String? = null

    @get:JsonProperty("rating")
    @set:JsonProperty("rating")
    @JsonProperty("rating")
    var rating: Double? = null

    @get:JsonProperty("rating_top")
    @set:JsonProperty("rating_top")
    @JsonProperty("rating_top")
    var ratingTop: Int? = null

    @get:JsonProperty("ratings")
    @set:JsonProperty("ratings")
    @JsonProperty("ratings")
    var ratings: List<GameGetRating>? = null

    @get:JsonProperty("reactions")
    @set:JsonProperty("reactions")
    @JsonProperty("reactions")
    var reactions: GameGetReactions? = null

    @get:JsonProperty("added")
    @set:JsonProperty("added")
    @JsonProperty("added")
    var added: Int? = null

    @get:JsonProperty("added_by_status")
    @set:JsonProperty("added_by_status")
    @JsonProperty("added_by_status")
    var addedByStatus: GameGetAddedByStatus? = null

    @get:JsonProperty("playtime")
    @set:JsonProperty("playtime")
    @JsonProperty("playtime")
    var playtime: Int? = null

    @get:JsonProperty("screenshots_count")
    @set:JsonProperty("screenshots_count")
    @JsonProperty("screenshots_count")
    var screenshotsCount: Int? = null

    @get:JsonProperty("movies_count")
    @set:JsonProperty("movies_count")
    @JsonProperty("movies_count")
    var moviesCount: Int? = null

    @get:JsonProperty("creators_count")
    @set:JsonProperty("creators_count")
    @JsonProperty("creators_count")
    var creatorsCount: Int? = null

    @get:JsonProperty("achievements_count")
    @set:JsonProperty("achievements_count")
    @JsonProperty("achievements_count")
    var achievementsCount: Int? = null

    @get:JsonProperty("parent_achievements_count")
    @set:JsonProperty("parent_achievements_count")
    @JsonProperty("parent_achievements_count")
    var parentAchievementsCount: Int? = null

    @get:JsonProperty("reddit_url")
    @set:JsonProperty("reddit_url")
    @JsonProperty("reddit_url")
    var redditUrl: String? = null

    @get:JsonProperty("reddit_name")
    @set:JsonProperty("reddit_name")
    @JsonProperty("reddit_name")
    var redditName: String? = null

    @get:JsonProperty("reddit_description")
    @set:JsonProperty("reddit_description")
    @JsonProperty("reddit_description")
    var redditDescription: String? = null

    @get:JsonProperty("reddit_logo")
    @set:JsonProperty("reddit_logo")
    @JsonProperty("reddit_logo")
    var redditLogo: String? = null

    @get:JsonProperty("reddit_count")
    @set:JsonProperty("reddit_count")
    @JsonProperty("reddit_count")
    var redditCount: Int? = null

    @get:JsonProperty("twitch_count")
    @set:JsonProperty("twitch_count")
    @JsonProperty("twitch_count")
    var twitchCount: Int? = null

    @get:JsonProperty("youtube_count")
    @set:JsonProperty("youtube_count")
    @JsonProperty("youtube_count")
    var youtubeCount: Int? = null

    @get:JsonProperty("reviews_text_count")
    @set:JsonProperty("reviews_text_count")
    @JsonProperty("reviews_text_count")
    var reviewsTextCount: Int? = null

    @get:JsonProperty("ratings_count")
    @set:JsonProperty("ratings_count")
    @JsonProperty("ratings_count")
    var ratingsCount: Int? = null

    @get:JsonProperty("suggestions_count")
    @set:JsonProperty("suggestions_count")
    @JsonProperty("suggestions_count")
    var suggestionsCount: Int? = null

    @get:JsonProperty("alternative_names")
    @set:JsonProperty("alternative_names")
    @JsonProperty("alternative_names")
    var alternativeNames: List<String>? = null

    @get:JsonProperty("metacritic_url")
    @set:JsonProperty("metacritic_url")
    @JsonProperty("metacritic_url")
    var metacriticUrl: String? = null

    @get:JsonProperty("parents_count")
    @set:JsonProperty("parents_count")
    @JsonProperty("parents_count")
    var parentsCount: Int? = null

    @get:JsonProperty("additions_count")
    @set:JsonProperty("additions_count")
    @JsonProperty("additions_count")
    var additionsCount: Int? = null

    @get:JsonProperty("game_series_count")
    @set:JsonProperty("game_series_count")
    @JsonProperty("game_series_count")
    var gameSeriesCount: Int? = null

    @get:JsonProperty("user_game")
    @set:JsonProperty("user_game")
    @JsonProperty("user_game")
    var userGame: Any? = null

    @get:JsonProperty("reviews_count")
    @set:JsonProperty("reviews_count")
    @JsonProperty("reviews_count")
    var reviewsCount: Int? = null

    @get:JsonProperty("saturated_color")
    @set:JsonProperty("saturated_color")
    @JsonProperty("saturated_color")
    var saturatedColor: String? = null

    @get:JsonProperty("dominant_color")
    @set:JsonProperty("dominant_color")
    @JsonProperty("dominant_color")
    var dominantColor: String? = null

    @get:JsonProperty("parent_platforms")
    @set:JsonProperty("parent_platforms")
    @JsonProperty("parent_platforms")
    var parentPlatforms: List<GameGetParentPlatformGroup>? = null

    @get:JsonProperty("platforms")
    @set:JsonProperty("platforms")
    @JsonProperty("platforms")
    var platforms: List<GameGetPlatformGroup>? = null

    @get:JsonProperty("stores")
    @set:JsonProperty("stores")
    @JsonProperty("stores")
    var stores: List<GameGetStoreGroup>? = null

    @get:JsonProperty("developers")
    @set:JsonProperty("developers")
    @JsonProperty("developers")
    var developers: List<GameGetDeveloper>? = null

    @get:JsonProperty("genres")
    @set:JsonProperty("genres")
    @JsonProperty("genres")
    var genres: List<GameGetGenre>? = null

    @get:JsonProperty("tags")
    @set:JsonProperty("tags")
    @JsonProperty("tags")
    var tags: List<GameGetTag>? = null

    @get:JsonProperty("publishers")
    @set:JsonProperty("publishers")
    @JsonProperty("publishers")
    var publishers: List<GameGetPublisher>? = null

    @get:JsonProperty("esrb_rating")
    @set:JsonProperty("esrb_rating")
    @JsonProperty("esrb_rating")
    var esrbRating: GameGetEsrbRating? = null

    @get:JsonProperty("clip")
    @set:JsonProperty("clip")
    @JsonProperty("clip")
    var clip: Any? = null

    @get:JsonProperty("description_raw")
    @set:JsonProperty("description_raw")
    @JsonProperty("description_raw")
    var descriptionRaw: String? = null

    @JsonIgnore
    private val additionalProperties: MutableMap<String, Any> = HashMap()
    @JsonAnyGetter
    fun getAdditionalProperties(): Map<String, Any> {
        return additionalProperties
    }

    @JsonAnySetter
    fun setAdditionalProperty(name: String, value: Any) {
        additionalProperties[name] = value
    }
}